pkg/controller/plan/kubevirt.go

```diff
@@ -1542,14 +1542,22 @@ func changeVmName(currName string, vmID string) string {
 	if nameExcludeChars.MatchString(newName) {
 		newName = nameExcludeChars.ReplaceAllString(newName, "")
 	}
-	for strings.HasPrefix(newName, "-") {
-		newName = newName[1:]
-	}
-	for strings.HasSuffix(newName, "-") {
-		newName = newName[:len(newName)-1]
-	}
+	newName = strings.Trim(newName, "-")
 	if len(newName) == 0 {
-		newName = "vm-" + vmID[:4]
+		newName = "vm-" + generateRandVmNameSuffix()
 	}
 	return newName
 }
+
+// Generates a random string of length six, consisting of lowercase letters and digits.
+func generateRandVmNameSuffix() string {
+	const charset = "abcdefghijklmnopqrstuvwxyz" + "0123456789"
+	source := rand.NewSource(time.Now().UTC().UnixNano())
+	seededRand := rand.New(source)
+
+	b := make([]byte, 6)
```

Comment:
shouldn't it be of length 4 as otherwise we may exceed the max length?

Reply:
yes, you're right, forgot about it (I can change it over there)
I've extended it to six for more unique naming, not sure if it's really needed...

Reply:
practically, I think it's fine to go with a random string of all those characters with length 4